### PR TITLE
add example Writing your own Route Enhancers

### PR DESCRIPTION
**Pull request by andrey1s**

### CODE REVIEW

File: bundles/routing/dynamic_customize.rst [hunk 1]
```diff
@@ -22,12 +22,57 @@ Writing your own Route Enhancers
...
+
+.. code-block:: php
+
+    // src/AppBundle/Routing/Enhancer/SimpleEnhancer.php
+    namespace AppBundle\Routing\Enhancer;
+
+    use Symfony\Cmf\Component\Routing\Enhancer\RouteEnhancerInterface;
+
+    class SimpleEnhancer implements RouteEnhancerInterface
+    {
```
Comment (dbu) [inline bundles/routing/dynamic_customize.rst]: this code would be invalid, we should add a method that does something trivial to make the example complete.

**Comment by dbu** [comment]
great, thanks! lets complete this with implementing a method in the enhancer that does something simple so its self-contained.

File: bundles/routing/dynamic_customize.rst [hunk 2]
```diff
@@ -22,12 +22,62 @@ Writing your own Route Enhancers
...
+    namespace AppBundle\Routing\Enhancer;
+
+    use Symfony\Cmf\Component\Routing\Enhancer\RouteEnhancerInterface;
+    use Symfony\Component\HttpFoundation\Request;
+
+    class SimpleEnhancer implements RouteEnhancerInterface
+    {
+        public function enhance(array $defaults, Request $request)
+        {
+            //own logic
```
Comment (WouterJ) [inline bundles/routing/dynamic_customize.rst]: Let's show that one has to return the new defaults array like this:
```php
public function enhance(array $defaults, Request $request)
{
    // ... tweak the $defaults array

    return $defaults;
}
```

File: bundles/routing/dynamic_customize.rst [hunk 3]
```diff
@@ -22,12 +22,62 @@ Writing your own Route Enhancers
...
 
+.. configuration-block::
+
+    .. code-block:: yaml
+
+        services:
+            app.routing.enhancer.simple:
+                class: AppBundle\Routing\Enhancer\SimpleEnhancer
+                tags:
+                    -  { name: dynamic_router_route_enhancer priority: 10 }
```
Comment (WouterJ) [inline bundles/routing/dynamic_customize.rst]: missing `,` before `priority`

File: bundles/routing/dynamic_customize.rst [hunk 4]
```diff
@@ -22,12 +22,62 @@ Writing your own Route Enhancers
...
+                </service>
+            </services>
+        </container>
+
+    .. code-block:: php
+
+        use Symfony\Component\DependencyInjection\Definition;
+
+        $definitionSendmail = new Definition('AppBundle\Routing\Enhancer\SimpleEnhancer');
+        $definitionSendmail->addTag('dynamic_router_route_enhancer',array('priority' => 10));
```
Comment (WouterJ) [inline bundles/routing/dynamic_customize.rst]: the variable seems to be a copy-paste mistake. Let's rename it to simply `$definition`.

**Comment by dbu** [comment]
looks good to me. @WouterJ see something that needs to be fixed?
@andrey1s may i ask you to squash the commits into one?

**Comment by WouterJ** [comment]
Thank you @andrey1s! I've merged your pull request into all versions of the documentation and did some minor formatting changes in b586f62

**Pull request closed without merging**